munit/shared/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -115,14 +115,19 @@ class MUnitRunner(val cls: Class[_ <: Suite], suite: Suite)
         val body =
           if (!beforeAll.isSuccess) Future.successful(Nil)
           else sequenceFutures(munitTests.iterator.map(runTest(notifier, _)))
-        body.transform { res =>
-          runAfterAll(notifier, beforeAll)
-          res.failed.foreach(ex =>
-            fireFailedHiddenTest(notifier, "unexpected error running tests", ex)
-          )
-          notifier.fireTestSuiteFinished(description)
-          util.Success(())
-        }
+        body.transformWith(res =>
+          runAfterAll(notifier, beforeAll).map { _ =>
+            res.failed.foreach(ex =>
+              fireFailedHiddenTest(
+                notifier,
+                "unexpected error running tests",
+                ex,
+              )
+            )
+            notifier.fireTestSuiteFinished(description)
+            util.Success(())
```

Comment:
with .map, this could be `()`, wouldn't need to return a Try. but it should have been .transform, just like before.